sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -495,15 +495,16 @@ private[hive] class HiveClientImpl(
     shim.dropTable(client, dbName, tableName, true, ignoreIfNotExists, purge)
   }
 
-  override def alterTable(tableName: String, table: CatalogTable): Unit = withHiveState {
+  override def alterTable(dbName: String, tableName: String, table: CatalogTable): Unit =
+      withHiveState {
```

Comment:
```Scala
override def alterTable(
    dbName: String,
    tableName: String,
    table: CatalogTable): Unit = withHiveState {
```

Reply:
This is not right. We should remove `tableName `.

Reply:
Let me change the impl of HiveClientImpl. Then, you can work on it.

Reply:
What's needed to change? Why can't the author works on it?

Reply:
https://github.com/apache/spark/pull/19104  

Need to write a few test cases for it. It might take multiple passes for it.